apps/crn-server/src/controllers/research-output.controller.ts

```diff
@@ -144,6 +146,20 @@ export default class ResearchOutputController {
     const currentResearchOutput =
       await this.researchOutputDataProvider.fetchById(id);
 
+    let version: ResearchOutputVersionPostRequest | undefined;
+    if (researchOutputUpdateData.createVersion) {
+      version = {
+        title: currentResearchOutput?.title || '',
+        link: currentResearchOutput?.link,
+        type: currentResearchOutput?.type,
+        documentType:
+          currentResearchOutput?.documentType &&
+          isResearchOutputDocumentType(currentResearchOutput?.documentType)
+            ? currentResearchOutput?.documentType
+            : 'Grant Document',
+      };
+    }
+
```

Comment:
i think the flow of the code would be better if this code was below the  `currentResearchOutput` check. 

Also should we be checking the RO is published before creating the version